be1-go/hub/standard_hub/mod.go

```diff
@@ -117,23 +117,27 @@ func NewHub(pubKeyOwner kyber.Point, clientServerAddress string, serverServerAdd
 		clientServerAddress: clientServerAddress,
 		serverServerAddress: serverServerAddress,
 		messageChan:         make(chan socket.IncomingMessage),
-		channelByID:         state.NewChannelsById(),
-		closedSockets:       make(chan string),
-		pubKeyOwner:         pubKeyOwner,
-		pubKeyServ:          pubServ,
-		secKeyServ:          secServ,
-		schemaValidator:     schemaValidator,
-		stop:                make(chan struct{}),
-		workers:             semaphore.NewWeighted(numWorkers),
-		log:                 log,
-		laoFac:              laoFac,
-		serverSockets:       channel.NewSockets(),
-		hubInbox:            *inbox.NewInbox(rootChannel),
-		rootInbox:           *inbox.NewInbox(rootChannel),
-		queries:             state.NewQueries(),
-		messageIdsByChannel: state.NewIdsByChannel(),
-		peers:               state.NewPeers(),
-		blacklist:           make([]string, 0),
+		channelByID: state.Channels{
+			ThreadSafeMap: state.NewThreadSafeMap[string, channel.Channel](),
+		},
+		closedSockets:   make(chan string),
+		pubKeyOwner:     pubKeyOwner,
+		pubKeyServ:      pubServ,
+		secKeyServ:      secServ,
+		schemaValidator: schemaValidator,
+		stop:            make(chan struct{}),
+		workers:         semaphore.NewWeighted(numWorkers),
+		log:             log,
+		laoFac:          laoFac,
+		serverSockets:   channel.NewSockets(),
+		hubInbox:        *inbox.NewInbox(rootChannel),
+		rootInbox:       *inbox.NewInbox(rootChannel),
+		queries:         state.NewQueries(),
+		messageIdsByChannel: state.MessageIds{
```

Comment:
you could also create a utility function `state.NewMessageIdsMap()` to encapsulate this.